### PR TITLE
Add Sphinx guide

### PR DESCRIPTION
**Pull request by luke-a-williams**
## Guide outcomes - the PR should cover the following content
- What Sphinx even is
- To understand potential use cases of this setup, as docstrings aren't for everyone!
- Launch a local Sphinx documentation server that dynamically updates in real time
- Sphinx configuration understands markdown, and all documentation files exist as markdown and not ReStructuredText.
- Sphinx configuration allows dynamic reading of Docstrings, of Google or Numpy standard format.
- How to handle Sphinx within version control.
- Local configuration is ready for potential deployment into CI/CD pipelines with proper routing.
- Demonstrate how to interact with the configuration to change the theme, and use extensions.

### CODE REVIEW

File: guides/sphinx_setup.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+- Sphinx configuration understands markdown, and all documentation files exist as markdown and not ReStructuredText.
+- Sphinx configuration allows dynamic reading of Docstrings, of Google or Numpy standard format.
+- How to handle Sphinx within version control.
+- Local configuration is ready for potential deployment into CI/CD pipelines with proper routing.
+- Demonstrate how to interact with the configuration to change the theme, and use extensions.
+- To understand potential use cases of this setup, as docstrings aren't for everyone!
+
+# Introduction - What is Sphinx?
+
+At it's core [Sphinx](https://www.sphinx-doc.org/en/master/index.html) is a documentation generator that uses reStructuredText or Markdown to create static documentation websites. It makes it easy to create intelligent and beautiful documentation for not just python projects, and can be embedded in many existing project setups.
```
Comment (tf75) [inline guides/sphinx_setup.md:16]: what other projects can you use sphinx for?

File: guides/sphinx_setup.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+I hosted a Data CoP that talked about `Sphinx`, and gave some examples of potential alternatives and competitors. If you want to see this CoP you can find the recording information in our [Facilitators sheet](https://docs.google.com/spreadsheets/d/1C4n1miK6Xaa5CHkZmjSQyBRtSm15PyyUpoCk_SzM0NE/edit?usp=sharing).
+
+## When might you want to use Sphinx?
+This section will give an insight into when it might be used, and also what other tools exist in this realm.
+
+### Examples of potential use cases
+Docstrings are not everyone's cup of tea. **But you also don't need Docstrings to use Sphinx**. Sphinx has some dynamic powers to reduce the amount of time you spend writing documentation that surrounds docstrings, sure, but since it is built off of Markdown in a `docs` directory, it can also set the stage for holding documentation in READMEs close to your code and thus within the same version control when you come to making PRs for code changes.
+
+Here are some potential project scenarios where Sphinx might be a good idea:
+- The [Skills for Care workstream](https://github.com/NMDSdevopsServiceAdm/DataEngineering/)
```
Comment (tf75) [inline guides/sphinx_setup.md:28]: presume public repo, if so fine to include

Comment (tf75) [reply guides/sphinx_setup.md:28]: but also seems a bit random as this isn't a project scenario this is an example of implementation. Maybe think about changing where this features, or just remove

File: guides/sphinx_setup.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+- When there is a project whose non technical stakeholders want better visibility of what the code is doing.
+- There is a need for documentation to see which code is responsible for specific parts of the pipeline, and this is particularly relevant if there are ML components to the pipeline stored in the same part of the code base.
+- You have multiple components to your project, such as an API front end as well as a webscreen that you want to signpost people toward when they visit your repo.
+- You don't have a front end development team and want easy rendering of static web-page documentation for any aspect of your project.
+- You have graphics or visuals like C4 diagrams close to your code you want to showcase somewhere.
+- You are on a workstream where there is some to no documentation, and you want to incentivise better documentation practises. You can use Sphinx to encourage docstring usage or even better functional practises if there is documentation that can build from it - i.e. do not repeat yourself practises translating across into docs.
```
Comment (tf75) [inline guides/sphinx_setup.md:34]: I just put DRY reads better

File: guides/sphinx_setup.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+- Local configuration is ready for potential deployment into CI/CD pipelines with proper routing.
+- Demonstrate how to interact with the configuration to change the theme, and use extensions.
+- To understand potential use cases of this setup, as docstrings aren't for everyone!
+
+# Introduction - What is Sphinx?
+
+At it's core [Sphinx](https://www.sphinx-doc.org/en/master/index.html) is a documentation generator that uses reStructuredText or Markdown to create static documentation websites. It makes it easy to create intelligent and beautiful documentation for not just python projects, and can be embedded in many existing project setups.
+[Sphinx](https://www.sphinx-doc.org/en/master/index.html) can look quite intimidating to a first-time user - it is packed full of features and has a large community base with plugins and themes galore
+
+I hosted a Data CoP that talked about `Sphinx`, and gave some examples of potential alternatives and competitors. If you want to see this CoP you can find the recording information in our [Facilitators sheet](https://docs.google.com/spreadsheets/d/1C4n1miK6Xaa5CHkZmjSQyBRtSm15PyyUpoCk_SzM0NE/edit?usp=sharing).
```
[the remark this answers is not in the document]
Comment (luke-a-williams) [reply guides/sphinx_setup.md:19]: Good shout, the doc is locked down to MadeTech but yes that makes way more sense in context

**Review by tf75 — APPROVED** [review]

**Pull request merged**